### ISSUE TITLE
Graph Visualization Issue

### ISSUE THREAD

**Issue by Nico-Strecker** [tensorflow/tensorboard#6989]
## Environment information (required)

Please run `diagnose_tensorboard.py` (link below) in the same
environment from which you normally run TensorFlow/TensorBoard, and
paste the output here:

/tmp/ipykernel_4586/3463113833.py:32: DeprecationWarning: 'pipes' is deprecated and slated for removal in Python 3.13
  import pipes
2025-04-24 00:31:26,243 - INFO - diagnose_tensorboard.py source unavailable

### Diagnostics

<details>
<summary>Diagnostics output</summary>

``````
--- check: autoidentify
INFO: diagnose_tensorboard.py source unavailable

2025-04-24 00:31:26,244 - INFO - sys.version_info: sys.version_info(major=3, minor=11, micro=11, releaselevel='final', serial=0)


--- check: general
INFO: sys.version_info: sys.version_info(major=3, minor=11, micro=11, releaselevel='final', serial=0)

2025-04-24 00:31:26,246 - INFO - os.name: posix

INFO: os.name: posix

2025-04-24 00:31:26,247 - INFO - os.uname(): posix.uname_result(sysname='Linux', nodename='7b192227c071', release='5.15.167.4-microsoft-standard-WSL2', version='#1 SMP Tue Nov 5 00:21:55 UTC 2024', machine='x86_64')

INFO: os.uname(): posix.uname_result(sysname='Linux', nodename='7b192227c071', release='5.15.167.4-microsoft-standard-WSL2', version='#1 SMP Tue Nov 5 00:21:55 UTC 2024', machine='x86_64')

2025-04-24 00:31:26,249 - INFO - sys.getwindowsversion(): N/A

INFO: sys.getwindowsversion(): N/A

--- check: package_management

2025-04-24 00:31:26,553 - INFO - has conda-meta: False

INFO: has conda-meta: False

2025-04-24 00:31:26,555 - INFO - $VIRTUAL_ENV: None

INFO: $VIRTUAL_ENV: None

--- check: installed_packages

2025-04-24 00:31:26,897 - INFO - installed: tensorboard==2.19.0

INFO: installed: tensorboard==2.19.0

2025-04-24 00:31:26,899 - INFO - installed: tensorflow==2.19.0

INFO: installed: tensorflow==2.19.0

2025-04-24 00:31:26,900 - WARNING - no installation among: ['tensorflow-estimator', 'tensorflow-estimator-2.0-preview', 'tf-estimator-nightly']

WARNING: no installation among: ['tensorflow-estimator', 'tensorflow-estimator-2.0-preview', 'tf-estimator-nightly']

2025-04-24 00:31:26,901 - INFO - installed: tensorboard-data-server==0.7.2

INFO: installed: tensorboard-data-server==0.7.2

2025-04-24 00:31:26,903 - INFO - tensorboard.version.VERSION: '2.19.0'


--- check: tensorboard_python_version
INFO: tensorboard.version.VERSION: '2.19.0'

2025-04-24 00:31:26,905 - INFO - tensorflow.__version__: '2.19.0'


--- check: tensorflow_python_version
INFO: tensorflow.__version__: '2.19.0'

2025-04-24 00:31:26,906 - INFO - tensorflow.__git_version__: 'v2.19.0-rc0-6-ge36baa30292'

INFO: tensorflow.__git_version__: 'v2.19.0-rc0-6-ge36baa30292'

2025-04-24 00:31:26,908 - INFO - data server binary: '/usr/local/lib/python3.11/dist-packages/tensorboard_data_server/bin/server'


--- check: tensorboard_data_server_version
INFO: data server binary: '/usr/local/lib/python3.11/dist-packages/tensorboard_data_server/bin/server'

2025-04-24 00:31:26,914 - INFO - data server binary version: b'rustboard 0.7.2'

INFO: data server binary version: b'rustboard 0.7.2'

2025-04-24 00:31:27,015 - INFO - which tensorboard: b'/usr/local/bin/tensorboard\n'


--- check: tensorboard_binary_path
INFO: which tensorboard: b'/usr/local/bin/tensorboard\n'

2025-04-24 00:31:27,018 - INFO - socket.getfqdn(): '7b192227c071'


--- check: addrinfos
socket.has_ipv6 = True
socket.AF_UNSPEC = <AddressFamily.AF_UNSPEC: 0>
socket.SOCK_STREAM = <SocketKind.SOCK_STREAM: 1>
socket.AI_ADDRCONFIG = <AddressInfo.AI_ADDRCONFIG: 32>
socket.AI_PASSIVE = <AddressInfo.AI_PASSIVE: 1>
Loopback flags: <AddressInfo.AI_ADDRCONFIG: 32>
Loopback infos: [(<AddressFamily.AF_INET: 2>, <SocketKind.SOCK_STREAM: 1>, 6, '', ('127.0.0.1', 0))]
Wildcard flags: <AddressInfo.AI_PASSIVE: 1>
Wildcard infos: [(<AddressFamily.AF_INET: 2>, <SocketKind.SOCK_STREAM: 1>, 6, '', ('0.0.0.0', 0)), (<AddressFamily.AF_INET6: 10>, <SocketKind.SOCK_STREAM: 1>, 6, '', ('::', 0, 0, 0))]

--- check: readable_fqdn
INFO: socket.getfqdn(): '7b192227c071'

2025-04-24 00:31:27,019 - INFO - directory: /tmp/.tensorboard-info


--- check: stat_tensorboardinfo
INFO: directory: /tmp/.tensorboard-info

2025-04-24 00:31:27,020 - INFO - os.stat(...): os.stat_result(st_mode=16895, st_ino=191515, st_dev=170, st_nlink=2, st_uid=0, st_gid=0, st_size=4096, st_atime=1745454017, st_mtime=1745454665, st_ctime=1745454665)

INFO: os.stat(...): os.stat_result(st_mode=16895, st_ino=191515, st_dev=170, st_nlink=2, st_uid=0, st_gid=0, st_size=4096, st_atime=1745454017, st_mtime=1745454665, st_ctime=1745454665)

2025-04-24 00:31:27,021 - INFO - mode: 0o40777

INFO: mode: 0o40777

2025-04-24 00:31:27,025 - INFO - tensorboard_roots (1): ['/usr/local/lib/python3.11/dist-packages']; bad_roots (0): []


--- check: source_trees_without_genfiles
INFO: tensorboard_roots (1): ['/usr/local/lib/python3.11/dist-packages']; bad_roots (0): []

--- check: full_pip_freeze

2025-04-24 00:31:27,305 - INFO - pip freeze --all:
absl-py==2.1.0
anyio==4.8.0
argon2-cffi==23.1.0
argon2-cffi-bindings==21.2.0
arrow==1.3.0
asttokens==3.0.0
astunparse==1.6.3
async-lru==2.0.4
attrs==25.1.0
babel==2.17.0
beautifulsoup4==4.13.3
bleach==6.2.0
blinker==1.4
certifi==2025.1.31
cffi==1.17.1
charset-normalizer==3.4.1
comm==0.2.2
contourpy==1.3.1
cryptography==3.4.8
cycler==0.12.1
dbus-python==1.2.18
debugpy==1.8.13
decorator==5.2.1
defusedxml==0.7.1
distro==1.7.0
distro-info==1.1+ubuntu0.2
executing==2.2.0
fastjsonschema==2.21.1
flatbuffers==25.2.10
fonttools==4.56.0
fqdn==1.5.1
frozendict==2.4.6
gast==0.6.0
google-pasta==0.2.0
grpcio==1.71.0
h11==0.14.0
h5py==3.13.0
httpcore==1.0.7
httplib2==0.20.2
httpx==0.28.1
humanize==4.12.2
idna==3.10
importlib-metadata==4.6.4
ipykernel==6.29.5
ipython==9.0.2
ipython_pygments_lexers==1.1.1
ipywidgets==8.1.5
isoduration==20.11.0
jedi==0.19.2
jeepney==0.7.1
Jinja2==3.1.6
joblib==1.4.2
json5==0.10.0
jsonpointer==3.0.0
jsonschema==4.23.0
jsonschema-specifications==2024.10.1
jupyter==1.1.1
jupyter-console==6.6.3
jupyter-events==0.12.0
jupyter-http-over-ws==0.0.8
jupyter-lsp==2.2.5
jupyter_client==8.6.3
jupyter_core==5.7.2
jupyter_server==2.15.0
jupyter_server_terminals==0.5.3
jupyterlab==4.3.5
jupyterlab_pygments==0.3.0
jupyterlab_server==2.27.3
jupyterlab_widgets==3.0.13
keras==3.9.0
keyring==23.5.0
kiwisolver==1.4.8
launchpadlib==1.10.16
lazr.restfulclient==0.14.4
lazr.uri==1.0.6
libclang==18.1.1
Markdown==3.7
markdown-it-py==3.0.0
MarkupSafe==3.0.2
matplotlib==3.10.1
matplotlib-inline==0.1.7
mdurl==0.1.2
mistune==3.1.2
ml_dtypes==0.5.1
more-itertools==8.10.0
multitasking==0.0.11
namex==0.0.8
nbclient==0.10.2
nbconvert==7.16.6
nbformat==5.10.4
nest-asyncio==1.6.0
notebook==7.3.2
notebook_shim==0.2.4
numpy==2.1.3
nvidia-cublas-cu12==12.5.3.2
nvidia-cuda-cupti-cu12==12.5.82
nvidia-cuda-nvcc-cu12==12.5.82
nvidia-cuda-nvrtc-cu12==12.5.82
nvidia-cuda-runtime-cu12==12.5.82
nvidia-cudnn-cu12==9.3.0.75
nvidia-cufft-cu12==11.2.3.61
nvidia-curand-cu12==10.3.6.82
nvidia-cusolver-cu12==11.6.3.83
nvidia-cusparse-cu12==12.5.1.3
nvidia-nccl-cu12==2.23.4
nvidia-nvjitlink-cu12==12.5.82
oauthlib==3.2.0
opt_einsum==3.4.0
optree==0.14.1
overrides==7.7.0
packaging==24.2
pandas==2.2.3
pandas-log==0.1.7
pandas_flavor==0.7.0
pandocfilters==1.5.1
parso==0.8.4
peewee==3.17.9
pexpect==4.9.0
pillow==11.1.0
pip==25.0.1
platformdirs==4.3.6
prometheus_client==0.21.1
prompt_toolkit==3.0.50
protobuf==5.29.3
psutil==7.0.0
ptyprocess==0.7.0
pure_eval==0.2.3
pycparser==2.22
Pygments==2.19.1
PyGObject==3.42.1
PyJWT==2.3.0
pyparsing==2.4.7
python-apt==2.4.0+ubuntu4
python-dateutil==2.9.0.post0
python-json-logger==3.3.0
pytz==2025.2
PyYAML==6.0.2
pyzmq==26.2.1
referencing==0.36.2
requests==2.32.3
rfc3339-validator==0.1.4
rfc3986-validator==0.1.1
rich==13.9.4
rpds-py==0.23.1
scikit-learn==1.6.1
scipy==1.15.2
SecretStorage==3.3.1
Send2Trash==1.8.3
setuptools==76.0.0
six==1.16.0
sniffio==1.3.1
soupsieve==2.6
stack-data==0.6.3
tensorboard==2.19.0
tensorboard-data-server==0.7.2
tensorflow==2.19.0
tensorflow-io-gcs-filesystem==0.37.1
termcolor==2.5.0
terminado==0.18.1
tf_keras==2.19.0
threadpoolctl==3.6.0
tinycss2==1.4.0
tornado==6.4.2
tqdm==4.67.1
traitlets==5.14.3
types-python-dateutil==2.9.0.20241206
typing_extensions==4.12.2
tzdata==2025.2
unattended-upgrades==0.1
uri-template==1.3.0
urllib3==2.3.0
wadllib==1.3.6
wcwidth==0.2.13
webcolors==24.11.1
webencodings==0.5.1
websocket-client==1.8.0
Werkzeug==3.1.3
wheel==0.45.1
widgetsnbextension==4.0.13
wrapt==1.17.2
xarray==2025.3.1
yfinance==0.2.55
zipp==1.0.0


INFO: pip freeze --all:
absl-py==2.1.0
anyio==4.8.0
argon2-cffi==23.1.0
argon2-cffi-bindings==21.2.0
arrow==1.3.0
asttokens==3.0.0
astunparse==1.6.3
async-lru==2.0.4
attrs==25.1.0
babel==2.17.0
beautifulsoup4==4.13.3
bleach==6.2.0
blinker==1.4
certifi==2025.1.31
cffi==1.17.1
charset-normalizer==3.4.1
comm==0.2.2
contourpy==1.3.1
cryptography==3.4.8
cycler==0.12.1
dbus-python==1.2.18
debugpy==1.8.13
decorator==5.2.1
defusedxml==0.7.1
distro==1.7.0
distro-info==1.1+ubuntu0.2
executing==2.2.0
fastjsonschema==2.21.1
flatbuffers==25.2.10
fonttools==4.56.0
fqdn==1.5.1
frozendict==2.4.6
gast==0.6.0
google-pasta==0.2.0
grpcio==1.71.0
h11==0.14.0
h5py==3.13.0
httpcore==1.0.7
httplib2==0.20.2
httpx==0.28.1
humanize==4.12.2
idna==3.10
importlib-metadata==4.6.4
ipykernel==6.29.5
ipython==9.0.2
ipython_pygments_lexers==1.1.1
ipywidgets==8.1.5
isoduration==20.11.0
jedi==0.19.2
jeepney==0.7.1
Jinja2==3.1.6
joblib==1.4.2
json5==0.10.0
jsonpointer==3.0.0
jsonschema==4.23.0
jsonschema-specifications==2024.10.1
jupyter==1.1.1
jupyter-console==6.6.3
jupyter-events==0.12.0
jupyter-http-over-ws==0.0.8
jupyter-lsp==2.2.5
jupyter_client==8.6.3
jupyter_core==5.7.2
jupyter_server==2.15.0
jupyter_server_terminals==0.5.3
jupyterlab==4.3.5
jupyterlab_pygments==0.3.0
jupyterlab_server==2.27.3
jupyterlab_widgets==3.0.13
keras==3.9.0
keyring==23.5.0
kiwisolver==1.4.8
launchpadlib==1.10.16
lazr.restfulclient==0.14.4
lazr.uri==1.0.6
libclang==18.1.1
Markdown==3.7
markdown-it-py==3.0.0
MarkupSafe==3.0.2
matplotlib==3.10.1
matplotlib-inline==0.1.7
mdurl==0.1.2
mistune==3.1.2
ml_dtypes==0.5.1
more-itertools==8.10.0
multitasking==0.0.11
namex==0.0.8
nbclient==0.10.2
nbconvert==7.16.6
nbformat==5.10.4
nest-asyncio==1.6.0
notebook==7.3.2
notebook_shim==0.2.4
numpy==2.1.3
nvidia-cublas-cu12==12.5.3.2
nvidia-cuda-cupti-cu12==12.5.82
nvidia-cuda-nvcc-cu12==12.5.82
nvidia-cuda-nvrtc-cu12==12.5.82
nvidia-cuda-runtime-cu12==12.5.82
nvidia-cudnn-cu12==9.3.0.75
nvidia-cufft-cu12==11.2.3.61
nvidia-curand-cu12==10.3.6.82
nvidia-cusolver-cu12==11.6.3.83
nvidia-cusparse-cu12==12.5.1.3
nvidia-nccl-cu12==2.23.4
nvidia-nvjitlink-cu12==12.5.82
oauthlib==3.2.0
opt_einsum==3.4.0
optree==0.14.1
overrides==7.7.0
packaging==24.2
pandas==2.2.3
pandas-log==0.1.7
pandas_flavor==0.7.0
pandocfilters==1.5.1
parso==0.8.4
peewee==3.17.9
pexpect==4.9.0
pillow==11.1.0
pip==25.0.1
platformdirs==4.3.6
prometheus_client==0.21.1
prompt_toolkit==3.0.50
protobuf==5.29.3
psutil==7.0.0
ptyprocess==0.7.0
pure_eval==0.2.3
pycparser==2.22
Pygments==2.19.1
PyGObject==3.42.1
PyJWT==2.3.0
pyparsing==2.4.7
python-apt==2.4.0+ubuntu4
python-dateutil==2.9.0.post0
python-json-logger==3.3.0
pytz==2025.2
PyYAML==6.0.2
pyzmq==26.2.1
referencing==0.36.2
requests==2.32.3
rfc3339-validator==0.1.4
rfc3986-validator==0.1.1
rich==13.9.4
rpds-py==0.23.1
scikit-learn==1.6.1
scipy==1.15.2
SecretStorage==3.3.1
Send2Trash==1.8.3
setuptools==76.0.0
six==1.16.0
sniffio==1.3.1
soupsieve==2.6
stack-data==0.6.3
tensorboard==2.19.0
tensorboard-data-server==0.7.2
tensorflow==2.19.0
tensorflow-io-gcs-filesystem==0.37.1
termcolor==2.5.0
terminado==0.18.1
tf_keras==2.19.0
threadpoolctl==3.6.0
tinycss2==1.4.0
tornado==6.4.2
tqdm==4.67.1
traitlets==5.14.3
types-python-dateutil==2.9.0.20241206
typing_extensions==4.12.2
tzdata==2025.2
unattended-upgrades==0.1
uri-template==1.3.0
urllib3==2.3.0
wadllib==1.3.6
wcwidth==0.2.13
webcolors==24.11.1
webencodings==0.5.1
websocket-client==1.8.0
Werkzeug==3.1.3
wheel==0.45.1
widgetsnbextension==4.0.13
wrapt==1.17.2
xarray==2025.3.1
yfinance==0.2.55
zipp==1.0.0

``````

</details>

https://raw.githubusercontent.com/tensorflow/tensorboard/master/tensorboard/tools/diagnose_tensorboard.py

For browser-related issues, please additionally specify:

  - Browser type and version (e.g., Chrome 64.0.3282.140):
  - Screenshot, if it’s a visual issue:

## Issue description

![Image](https://github.com/user-attachments/assets/6e881f85-e715-492a-8492-0882b0da87a9)

I dont have a bad connection nor do i have multiple graph defs.

In fact i even cleared my train folder but the issue persisted.

I am not sure what is causing the error i am fairly new to Tensorflow 

Could it be this layer despite defining an output shape?
```python
    title_item_gru = Lambda(
        lambda x: __import__("tensorflow").reshape(x, [-1, D, I, GRU_TITLE_UNITS]),
        output_shape=(D, I, GRU_TITLE_UNITS), # Explizite Ausgabeform (ohne Batch)
        name='title_reshape_unflat'
    )(title_word_gru_flat)
```
In my first run it worked but after loading the model after some epochs no graph is created 

I had to use __import__("tensorflow")... because tf.reshape didnt work when loading the model despite using `custom_object_scope` or `custom_objects=...`

**Comment by shijunliu-google** [comment]
We apologize that this project is currently in maintenance mode, and we don't have the immediate resources to address this issue. We recommend searching online for potential solutions.
